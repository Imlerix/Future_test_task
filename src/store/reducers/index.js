export { default as table } from './TableReducer';
