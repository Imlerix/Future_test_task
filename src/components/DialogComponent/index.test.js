import renderer from 'react-test-renderer'
import React from 'react';
import { Provider } from 'react-redux';
import { initStore } from '../../store';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import EnzymeToJson from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { DialogComponent } from './index';

Enzyme.configure({ adapter: new Adapter() });

let container = null;
const mockStore = initStore();

beforeEach(() => {
    // подготавливаем DOM-элемент, куда будем рендерить
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // подчищаем после завершения
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("SNAPSHOT renders correctly", () => {
    const subject = shallow(
        <DialogComponent open={true}
                         onClose={() => {}}/>
    );
    expect(EnzymeToJson(subject)).toMatchSnapshot();
});
