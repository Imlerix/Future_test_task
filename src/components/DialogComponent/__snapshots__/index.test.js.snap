// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT renders correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="simple-dialog-title"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    Добавить нового пользователя
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      key="0"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_id"
        error={false}
        fullWidth={true}
        helperText="Только цифры"
        id={0}
        label="ID"
        name="id"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="1"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_firstName"
        error={false}
        fullWidth={true}
        helperText="Только буквы без пробелов"
        id={1}
        label="Имя"
        name="firstName"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="2"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_lastName"
        error={false}
        fullWidth={true}
        helperText="Только буквы без пробелов"
        id={2}
        label="Фамилия"
        name="lastName"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="3"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_email"
        error={false}
        fullWidth={true}
        helperText="Например: qwerty@gmail.com"
        id={3}
        label="Эл. почта"
        name="email"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="4"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_phone"
        error={false}
        fullWidth={true}
        helperText="Например +79991235678 или 89991235678"
        id={4}
        label="Моб. телефон"
        name="phone"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Tooltip))
    aria-label="save"
    disableHoverListener={false}
    style={
      Object {
        "width": "100%",
      }
    }
    title="Сначала заполните поля!"
  >
    <div
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        disabled={true}
        fullWidth={true}
        onClick={[Function]}
        size="large"
        startIcon={<UNDEFINED />}
        variant="contained"
      >
        Добавить в таблицу
      </WithStyles(ForwardRef(Button))>
    </div>
  </WithStyles(ForwardRef(Tooltip))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="simple-dialog-title"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    Добавить нового пользователя
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      key="0"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_id"
        error={false}
        fullWidth={true}
        helperText="Только цифры"
        id={0}
        label="ID"
        name="id"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="1"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_firstName"
        error={false}
        fullWidth={true}
        helperText="Только буквы без пробелов"
        id={1}
        label="Имя"
        name="firstName"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="2"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_lastName"
        error={false}
        fullWidth={true}
        helperText="Только буквы без пробелов"
        id={2}
        label="Фамилия"
        name="lastName"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="3"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_email"
        error={false}
        fullWidth={true}
        helperText="Например: qwerty@gmail.com"
        id={3}
        label="Эл. почта"
        name="email"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="4"
    >
      <WithStyles(ForwardRef(TextField))
        className="DialogComponent_phone"
        error={false}
        fullWidth={true}
        helperText="Например +79991235678 или 89991235678"
        id={4}
        label="Моб. телефон"
        name="phone"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Tooltip))
    aria-label="save"
    disableHoverListener={false}
    style={
      Object {
        "width": "100%",
      }
    }
    title="Сначала заполните поля!"
  >
    <div
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        disabled={true}
        fullWidth={true}
        onClick={[Function]}
        size="large"
        startIcon={<UNDEFINED />}
        variant="contained"
      >
        Добавить в таблицу
      </WithStyles(ForwardRef(Button))>
    </div>
  </WithStyles(ForwardRef(Tooltip))>
</WithStyles(ForwardRef(Dialog))>
`;
